internal/pkg/manifest/svc.go

```diff
@@ -172,23 +196,57 @@ func (c *Count) Desired() (*int, error) {
 // Percentage represents a valid percentage integer ranging from 0 to 100.
 type Percentage int
 
+// ScalingConfigOrPercentage represents a resource that has autoscaling configurations or a default percentage.
+type ScalingConfigOrPercentage struct {
+	Value         *Percentage
+	ScalingConfig AdvancedScalingConfig // mutually exclusive with Value
+}
+
+// AdvancedScalingConfig represents advanced configurable options for a scaling policy.
+type AdvancedScalingConfig struct {
+	Value    *Percentage `yaml:"value"`
+	Cooldown Cooldown    `yaml:"cooldown"`
+}
+
+// Cooldown represents the autoscaling cooldown of resources.
+type Cooldown struct {
+	ScaleInCooldown  *time.Duration `yaml:"in"`
+	ScaleOutCooldown *time.Duration `yaml:"out"`
+}
+
 // AdvancedCount represents the configurable options for Auto Scaling as well as
 // Capacity configuration (spot).
 type AdvancedCount struct {
-	Spot         *int           `yaml:"spot"` // mutually exclusive with other fields
-	Range        Range          `yaml:"range"`
-	CPU          *Percentage    `yaml:"cpu_percentage"`
-	Memory       *Percentage    `yaml:"memory_percentage"`
-	Requests     *int           `yaml:"requests"`
-	ResponseTime *time.Duration `yaml:"response_time"`
-	QueueScaling QueueScaling   `yaml:"queue_delay"`
+	Spot         *int                      `yaml:"spot"` // mutually exclusive with other fields
+	Range        Range                     `yaml:"range"`
+	Cooldown     Cooldown                  `yaml:"cooldown"`
+	CPU          ScalingConfigOrPercentage `yaml:"cpu_percentage"`
+	Memory       ScalingConfigOrPercentage `yaml:"memory_percentage"`
+	Requests     *int                      `yaml:"requests"`
+	ResponseTime *time.Duration            `yaml:"response_time"`
+	QueueScaling QueueScaling              `yaml:"queue_delay"`
```

Comment:
We need cool down support for these scaling policies as well :)

Reply:
Will add these in a separate PR after the cooldown is working fully for the general, cpu, and memory fields.